zkevm-circuits/src/pi_circuit2.rs

```diff
@@ -595,26 +533,23 @@ impl<F: Field> PiCircuit<F> {
             _marker: PhantomData,
         }
     }
-
-    /// create a new PiCircuit with extra data
-    pub fn new_from_block_with_extra(block: &witness::Block<F>, taiko: &witness::Taiko) -> Self {
-        PiCircuit::new(PublicData::new(block, taiko))
-    }
 }
 
 impl<F: Field> SubCircuit<F> for PiCircuit<F> {
     type Config = PiCircuitConfig<F>;
 
     fn unusable_rows() -> usize {
-        2usize.pow(MIN_DEGREE as u32) - USED_ROWS + 2
+        // No column queried at more than 3 distinct rotations, so returns 6 as
+        // minimum unusable rows.
+        6
     }
 
     fn min_num_rows_block(_block: &witness::Block<F>) -> (usize, usize) {
         (USED_ROWS, USED_ROWS)
     }
 
     fn new_from_block(block: &witness::Block<F>) -> Self {
-        PiCircuit::new(PublicData::new(block, &witness::Taiko::default()))
+        PiCircuit::new(PublicData::new(block, &block.protocol_instance))
```

Comment:
We should call this circuit `TaikoPiCircuit`/`TaikoPiCircuitConfig` and put it in `taiko_pi_circuit.rs`, will be more clear like that.